module/builder/consensus/builder.go

```diff
@@ -246,6 +246,10 @@ func (b *Builder) BuildOn(parentID flow.Identifier, setter func(*flow.Header) er
 			return nil, fmt.Errorf("seal with empty state commitment: %v", irSeal.ID())
 		}
 		if irSeal2, found := byBlock[irSeal.Seal.BlockID]; found {
+			block, err := b.headers.ByBlockID(irSeal.Seal.BlockID)
+			if err != nil {
+				return nil, fmt.Errorf("could not retrieve block for seal: %w", err)
```

Comment:
```suggestion
				fmt.Println("could not retrieve block for seal", err)
				block := &flow.Header{}
```

If, for whatever reason, we can't get the block (i.e block not exist), I think we should still log the error, otherwise, the alert won't trigger.

Reply:
thanks -- great suggestion. 
I moved the entire block to construct the map `blockId -> seal` into its own function to reduce bloating of the main business logic. Included an error message at the point you suggested where the block height is not used (oterhwise, we will print the message with the default value of `0`)